### PR TITLE
PR: Use a better icon when no matches are found (Find/Replace widget)

### LINKED ISSUE #20255: Consider a different icon for "no matches" in Find/Replace
As a followup to PR #20049 , we might want to consider a different icon for "no matches" in the find/replace widget, and perhaps shift the "Bad Expression" icon to match. 

Currently, they use the following (per @ccordoba12 ):

**Warning**
warning icon

**Error**
error icon

This matches the icons used for warnings and errors in the Editor linting panel and Code Analysis pane, etc., treating no matches as a warning, and a regex syntax error as an error.

However, we might want to reconsider, as at least to me, the "warning" icon, :warning: i.e. "alert", to me implies there's something wrong with the expression I typed, whereas a red circled :x: is actually closer, if anything, to the symbology I'd expect if no matches were found, with the red X meaning "no" or "none".

No matches being found doesn't necessarily implied something was wrong with what the user typed, it instead might be either a positive or negative outcome for users. Quite often, I use it to check for uses of things I'm changing, have deprecated or I otherwise want to make sure don't exist in a certain file. Or, things I'm just not entirely sure whether they are in there or not, and just want to check.

Instead, to send a clearer and less ambiguous message, as well as avoid any confusion with the usage elsewhere with the "warning" and "error" semantics that this doesn't directly parallel (at least with "warning"), I'd suggest choosing a different icon to indicate "no match", such as a ~~yellow~~ orange or strong grey :no_entry: symbol, i.e. the Material Design icon "Do not disturb", or perhaps the :no_entry_sign: ("Block") icon, which carries more of the implication of nothing being found as opposed to being a warning/problem with the user's entry, as the parallel to the warning symbol elsewhere could imply.

image

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

- The current icon (the warning one) did not represent correctly the lack of matches after searching for text.

    **Before**

    imagen

    **After**

    image

- This is a follow-up of #20049.

### Issue(s) Resolved

Fixes #20255.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

**Review by dalthviz — COMMENTED** [review]
Thanks @ccordoba12 ! I think the new icon looks good. Also while checking this I got a traceback related to the findreplace code and the split editor/new window functionality:

```
Traceback (most recent call last):
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2265, in focus_changed
    self.refresh()
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2425, in refresh
    self.find_widget.set_editor(editor, refresh=False)
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\widgets\findreplace.py", line 452, in set_editor
    self.editor.textChanged.disconnect(self.update_matches)
RuntimeError: wrapped C/C++ object of type CodeEditor has been deleted
Traceback (most recent call last):
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2219, in current_changed
    self.find_widget.set_editor(editor, refresh=False)
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\widgets\findreplace.py", line 452, in set_editor
    self.editor.textChanged.disconnect(self.update_matches)
RuntimeError: wrapped C/C++ object of type CodeEditor has been deleted
Traceback (most recent call last):
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2265, in focus_changed
    self.refresh()
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2425, in refresh
    self.find_widget.set_editor(editor, refresh=False)
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\widgets\findreplace.py", line 452, in set_editor
    self.editor.textChanged.disconnect(self.update_matches)
RuntimeError: wrapped C/C++ object of type CodeEditor has been deleted
Traceback (most recent call last):
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2219, in current_changed
    self.find_widget.set_editor(editor, refresh=False)
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\widgets\findreplace.py", line 452, in set_editor
    self.editor.textChanged.disconnect(self.update_matches)
RuntimeError: wrapped C/C++ object of type CodeEditor has been deleted
```

Should that be solved in a different PR perhaps, maybe at #20221 ?

**Comment by ccordoba12** [comment]
Thanks @dalthviz for the review. I also encountered that problem and fixed it in PR #20221, specifically in commit [547348b](https://github.com/spyder-ide/spyder/pull/20221/commits/547348b962e3d677f9dc02fe71589da435aed20d) because it was catched by one of the new tests I added there.

**Review by CAM-Gerlach — COMMENTED** [review]
LGTM overall, though could you maybe reduce the padding between the icon and the "clear field" (delete) button to about to half the current value? It looks rather strange currently and takes up more of the field's space than necessary. Thanks!

**Review by dalthviz — APPROVED** [review]
Thanks for the info @ccordoba12 , then this LGTM :+1:

I will leave this approved but I think that @CAM-Gerlach comment makes sense too, so if it is possible/not to complicated maybe it could be worthy to try to do the change before merging this

**Comment by ccordoba12** [comment]
> LGTM overall, though could you maybe reduce the padding between the icon and the "clear field" (delete) button to about to half the current value?

As it's almost the case, it was not trivial to implement in Qt, but I managed to do it. See my original description for the updated screenshot.

@dalthviz, could you check that it looks ok on Windows and Mac? Thanks!

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, pending @dalthviz 's checks (I'm still on a trip right now, sorry)

**Pull request merged**
